src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -594,18 +621,39 @@ private async Task CheckJobCompletionStatus()
 
         private async Task<SearchResult> ExecuteReindexQueryAsync(ReindexJobQueryStatus queryStatus, bool countOnly, CancellationToken cancellationToken)
         {
+            SearchResultReindex searchResultReindex = GetSearchResultReindex(queryStatus.ResourceType);
             var queryParametersList = new List<Tuple<string, string>>()
             {
                 Tuple.Create(KnownQueryParameterNames.Count, _throttleController.GetThrottleBatchSize().ToString(CultureInfo.InvariantCulture)),
                 Tuple.Create(KnownQueryParameterNames.Type, queryStatus.ResourceType),
             };
 
+            // This should never be cosmos
+            if (searchResultReindex != null && searchResultReindex.CurrentResourceSurrogateId > 0)
+            {
+                // if queryStatus has value then it's a continuation otherwise first time for the resource type
+                long startResourceSurrogateId = Math.Max(queryStatus.StartResourceSurrogateId, searchResultReindex.CurrentResourceSurrogateId);
+
+                queryParametersList.AddRange(new[]
+                {
+                    Tuple.Create(KnownQueryParameterNames.EndSurrogateId, searchResultReindex.EndResourceSurrogateId.ToString()),
+                    Tuple.Create(KnownQueryParameterNames.StartSurrogateId, startResourceSurrogateId.ToString()),
+                    Tuple.Create(KnownQueryParameterNames.GlobalEndSurrogateId, "0"),
+                });
```

Comment:
Feeling like this would be a useful extension i.e. queryParametersList.AddSurrogateIdRange(min, max, global)

Reply:
Well after searching the code base there's only 1 location, the one you have highlighted, that only adds those 3 values. The other locations are adding 4 or 5 parameters. The closest candidate is ExportJobTask. Let me know what you think about it.